### PR TITLE
Print activation instructions for a venv after one has been created

### PR DESCRIPTION
**Pull request by 0v00**
## Summary

- Add prompts for users to activate the `.venv` after it has been created.
- Closes [https://github.com/astral-sh/uv/issues/1326](https://github.com/astral-sh/uv/issues/1326)

If the platform os matches `Os::Windows` then we provide the following prompt:
```rust
writeln!(
   printer,
   "Activate with .\\{}\\Scripts\\activate.ps1",
   path.normalized_display().cyan()
)
```

Otherwise we provide the following prompt:
```rust
writeln!(
  printer,
  "Activate with source {}/bin/activate",
  path.normalized_display().cyan()
)
```

I've added additional filters in `crates/uv/tests/venv.rs` so there shouldn't be issues on any of the CI runs (not sure what some of the Windows paths could look like, so these filters might need to be changed):
```rust
        (
            r".*?\\\.venv\\Scripts\\activate\.ps1", 
            "Activate with source /home/ferris/project/.venv/bin/activate"
        )
```

## Test Plan

- Ran tests in `crates/uv/tests/venv.rs` and updated snapshots.
- Ran `uv venv` in a Docker container to test that the output is correct (running Ubuntu)

### CODE REVIEW

File: crates/uv/src/commands/venv.rs [hunk 1]
```diff
@@ -188,5 +189,22 @@ async fn venv_impl(
         }
     }
 
+    match &platform.os() {
+        Os::Windows => {
```
Comment (T-256) [inline crates/uv/src/commands/venv.rs:193]: what about cmd users? 
- https://github.com/astral-sh/uv/pull/1523

File: crates/uv/src/commands/venv.rs [hunk 2]
```diff
@@ -188,5 +189,22 @@ async fn venv_impl(
...
+            writeln!(
+                printer,
+                "Activate with .\\{}\\Scripts\\activate.ps1",
+                path.normalized_display().cyan()
+            )
+        },
+        _ => {
+            writeln!(
+                printer,
+                "Activate with source {}/bin/activate",
```
Comment (T-256) [inline crates/uv/src/commands/venv.rs:203]:
```suggestion
                "Activate with: source {}/bin/activate",
```
or put command in quotations for improve readability.

File: crates/uv/src/commands/venv.rs (hunk 1 above)
Comment (zanieb) [inline crates/uv/src/commands/venv.rs:193]: Note we usually use a `cfg` for this, e.g. (from elsewhere)

```rust
    let bin_name = if cfg!(unix) {
        "bin"
    } else if cfg!(windows) {
        "Scripts"
    } else {
        unimplemented!("Only Windows and Unix are supported")
    };
```

Comment (zanieb) [reply crates/uv/src/commands/venv.rs:193]: I wonder if we can detect the current terminal type in Windows...

Comment (zanieb) [reply crates/uv/src/commands/venv.rs:193]: (or if we should just show both?

Comment (0v00) [reply crates/uv/src/commands/venv.rs:193]: Not sure if we can detect the current terminal type in Windows correctly/in a reliable way, would showing both like this be enough?

```bash
Activate with:
- Powershell: .\.venv\Scripts\activate.ps1
- CMD: .\.venv\Scripts\activate.bat
```

Comment (0v00) [reply crates/uv/src/commands/venv.rs:193]: Do we want to include `unimplemented!("Only Windows and Unix are supported")` as well for consistency?

File: crates/uv/src/commands/venv.rs (hunk 2 above)
Comment (0v00) [reply crates/uv/src/commands/venv.rs]: Good suggestion, have added this in.

File: crates/uv/src/commands/venv.rs [hunk 3]
```diff
@@ -189,22 +188,23 @@ async fn venv_impl(
...
-                path.normalized_display().cyan()
-            )
-        }
-    }.into_diagnostic()?;
+    if cfg!(windows) {
+        writeln!(
+            printer,
+            "Activate with:\n\
+            - Powershell: .\\{0}\\Scripts\\activate.ps1\n\
+            - CMD: .\\{0}\\Scripts\\activate.bat",
```
Comment (T-256) [inline crates/uv/src/commands/venv.rs:196]: I usually don't write `.\` when I'm in CMD environment.

Comment (0v00) [reply crates/uv/src/commands/venv.rs]: removed `.\` from the prompt

**Review by T-256 — APPROVED** [review]

File: crates/uv/src/commands/venv.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv/src/commands/venv.rs]: @AlexWaygood / @MichaReiser can you take over review of this?

Comment (AlexWaygood) [reply crates/uv/src/commands/venv.rs]: I believe typing `.\.venv\Scripts\activate` (omitting the file extension) _should_ actually work on both PowerShell and CMD

Comment (0v00) [reply crates/uv/src/commands/venv.rs]: Nice, that's much simpler

File: crates/uv/src/commands/venv.rs [hunk 4]
```diff
@@ -195,5 +195,21 @@ async fn venv_impl(
         }
     }
 
+    if cfg!(windows) {
+        writeln!(
+            printer,
```
[the remark this answers is not in the document]
Comment (0v00) [reply crates/uv/src/commands/venv.rs:200]: Good idea, added this in

**Pull request merged**